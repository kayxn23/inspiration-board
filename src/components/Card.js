import React, { Component } from 'react';
import PropTypes from 'prop-types';
import emoji from 'emoji-dictionary';

import './Card.css';

class Card extends Component {
  render() {
    // console.log('card props here:',this.props);

    return (
      <div className="card">
        Card
        <section className="card__content-text">
        <h3>{this.props.text}</h3>
        </section>

        <section className="card__content-emoji">
        {this.props.emoji &&  <p>{emoji.getUnicode(this.props.emoji) ? emoji.getUnicode(this.props.emoji) : this.props.emoji}</p>}
        </section>

      </div>
    )
  }
}

Card.propTypes = {
  text: PropTypes.string.isRequired,
  emoji: PropTypes.string,
  id: PropTypes.number
};

export default Card;
