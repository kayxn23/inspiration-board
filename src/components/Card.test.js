import React from 'react';
import Card from './Card';
import {shallow} from 'enzyme' //shallow is deconstructor, going into enzyme and pulling shallow from it

  describe('Card', () => {
    test('it matches an existing snapshop', () => {
      //make an instance of card see if it
      //looks like something it should
      const wrapper = shallow( <Card
        id={1}
        text="dumpster fire!"
        emoji="dumpster-fire-emoji"
  />);
    expect(wrapper).toMatchSnapshot();
  });
});
