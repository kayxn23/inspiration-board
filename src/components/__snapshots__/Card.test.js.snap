// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card it matches an existing snapshop 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    emoji="dumpster-fire-emoji"
    id={1}
    text="dumpster fire!"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <section
        className="card__content"
      >
        <section
          className="card__content-text"
        >
          <h3>
            dumpster fire!
          </h3>
        </section>
        <section
          className="card__content-emoji"
        >
          <p>
            dumpster-fire-emoji
          </p>
        </section>
        <button
          aria-label="Close"
          className="card__delete"
          onClick={[Function]}
          type="button"
        >
          <span
            aria-hidden="true"
          >
            ×
          </span>
        </button>
      </section>,
      "className": "card",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <section
            className="card__content-text"
          >
            <h3>
              dumpster fire!
            </h3>
          </section>,
          <section
            className="card__content-emoji"
          >
            <p>
              dumpster-fire-emoji
            </p>
          </section>,
          <button
            aria-label="Close"
            className="card__delete"
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden="true"
            >
              ×
            </span>
          </button>,
        ],
        "className": "card__content",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h3>
              dumpster fire!
            </h3>,
            "className": "card__content-text",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "dumpster fire!",
            },
            "ref": null,
            "rendered": "dumpster fire!",
            "type": "h3",
          },
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p>
              dumpster-fire-emoji
            </p>,
            "className": "card__content-emoji",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "dumpster-fire-emoji",
            },
            "ref": null,
            "rendered": "dumpster-fire-emoji",
            "type": "p",
          },
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-label": "Close",
            "children": <span
              aria-hidden="true"
            >
              ×
            </span>,
            "className": "card__delete",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "aria-hidden": "true",
              "children": "×",
            },
            "ref": null,
            "rendered": "×",
            "type": "span",
          },
          "type": "button",
        },
      ],
      "type": "section",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <section
          className="card__content"
        >
          <section
            className="card__content-text"
          >
            <h3>
              dumpster fire!
            </h3>
          </section>
          <section
            className="card__content-emoji"
          >
            <p>
              dumpster-fire-emoji
            </p>
          </section>
          <button
            aria-label="Close"
            className="card__delete"
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden="true"
            >
              ×
            </span>
          </button>
        </section>,
        "className": "card",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <section
              className="card__content-text"
            >
              <h3>
                dumpster fire!
              </h3>
            </section>,
            <section
              className="card__content-emoji"
            >
              <p>
                dumpster-fire-emoji
              </p>
            </section>,
            <button
              aria-label="Close"
              className="card__delete"
              onClick={[Function]}
              type="button"
            >
              <span
                aria-hidden="true"
              >
                ×
              </span>
            </button>,
          ],
          "className": "card__content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h3>
                dumpster fire!
              </h3>,
              "className": "card__content-text",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "dumpster fire!",
              },
              "ref": null,
              "rendered": "dumpster fire!",
              "type": "h3",
            },
            "type": "section",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <p>
                dumpster-fire-emoji
              </p>,
              "className": "card__content-emoji",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "dumpster-fire-emoji",
              },
              "ref": null,
              "rendered": "dumpster-fire-emoji",
              "type": "p",
            },
            "type": "section",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "aria-label": "Close",
              "children": <span
                aria-hidden="true"
              >
                ×
              </span>,
              "className": "card__delete",
              "onClick": [Function],
              "type": "button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "aria-hidden": "true",
                "children": "×",
              },
              "ref": null,
              "rendered": "×",
              "type": "span",
            },
            "type": "button",
          },
        ],
        "type": "section",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
